pkg/mesh/mesh.go

```diff
@@ -245,13 +247,14 @@ func (m *Mesh) Run() error {
 			m.syncNodes(ne)
 		case pe = <-pw:
 			m.syncPeers(pe)
-		case <-t.C:
+		case <-checkIn.C:
 			m.checkIn()
```

Comment:
Does this need `checkIn.Reset(checkInPeriod)`?

Reply:
Also, it seems like this is written in a way that only allows the check in to happen once before being removed from other nodes. You kind of want something like `if m.checkIn() { reset(30s) } else { reset(2s) }`.

Reply:
:see_no_evil: totally, this needs a reset. It would be great to have a nicer back-off and retry mechanism for the check-ins, agreed. This can be a follow-up IMO since it is a new feature.